### PR TITLE
feat: add getIdempotencyKey utility

### PR DESCRIPTION
**Pull request by michealroberts**
feat: add getIdempotencyKey utility

### CODE REVIEW

File: pkg/xero/base.go [hunk 1]
```diff
@@ -32,6 +37,23 @@ func (m HTTPMethod) String() string {
 	return "unknown"
 }
 
+func getIdempotencyKey(tenantId string, entity string) string {
```
Comment (Drache93) [inline pkg/xero/base.go]: <3

File: pkg/xero/base.go [hunk 2]
```diff
@@ -32,6 +37,23 @@ func (m HTTPMethod) String() string {
...
+	// Generate a ULID based on the current timestamp
+	now := time.Now().UTC()
+	// Generate a new entropy source:
+	entropy := ulid.Monotonic(rand.Reader, 0)
+	// Generate a ULID:
+	id, err := ulid.New(ulid.Timestamp(now), entropy)
+
+	// Oooh dear :(
+	if err != nil {
+		log.Fatalf("failed to generate ULID: %v", err)
```
Comment (Drache93) [inline pkg/xero/base.go]: Won't this kill who ever is trying to use it? Maybe return error instead so upstream can handle the fatal?

**Review by Drache93 — CHANGES_REQUESTED** [review]

File: pkg/xero/base.go (hunk 2 above)
Comment (michealroberts) [reply pkg/xero/base.go]: Yeh good shout.

Comment (michealroberts) [reply pkg/xero/base.go]: Amended ✅

Comment (michealroberts) [reply pkg/xero/base.go]: For the minute, if this function returns an error, I don't set the header.

**Review by Drache93 — APPROVED** [review]

**Pull request merged**